metamorph/Server.go

```diff
@@ -224,6 +228,10 @@ func (s *Server) PutTransactions(ctx context.Context, req *metamorph_api.Transac
 	// if not we store the transaction data and set the transaction status in response array to - STORED
 	requests := make([]*store.StoreData, 0, len(req.Transactions))
 	for ind, req := range req.Transactions {
+		if _, err := url.ParseRequestURI(req.CallbackUrl); req.CallbackUrl != "" && err != nil {
```

Comment:
If err is not nil it automatically means that req.CallbackUrl is not empty (so don't need to check) but that's ok

Reply:
No, because if the callback is empty (which is a valid input), then there will also be an error. Therefor we must return an error only if `err != nil && req.CallbackUrl != ""`